### DISCUSSION TITLE
What's the usual way to load one of these packages?

### DISCUSSION THREAD

**Discussion by Folaht** [lumen-oss/rocks.nvim#593] [Q&A] [answered]
I'm beginning to figure out the setup I need for my nvim configuration:

```
~/.config/nvim
├── init.lua
├── plugin
│   ├── colorscheme.lua
│   └── keymap
│       └── dvorak.lua
└── rocks.toml
```

`init.lua` for loading rocks.  
`rocks.toml` for installing packages.  
`plugin` folder where all files, including in subfolders 
are automatically loaded by nvim,  
thus ideal for loading packages, color schemes
and custom configurations.
`colorscheme.lua` One line loading my preferred colorscheme.
`dvorak.lua` My custom keymappings, 
as I prefer orthogonal dvorak keyboards over qwerty.  

It's obvious to me now that I will need a `plugins.lua` file in the `plugin` folder  
that will load all the packages that I've installed,   
but I'm still at a loss as to how,  
like for example how would one load `mason.nvim`?

*init.lua (manual version)*
```
local rocks_config = {
    rocks_path = vim.fn.stdpath("data") .. "/rocks",
}
...
vim.opt.runtimepath:append(vim.fs.joinpath(rocks_config.rocks_path, "lib", "luarocks", "rocks-5.1", "rocks.nvim", "*"))
```

*rocks.toml*
```
...
[rocks]

# List of Neovim plugins to install alongside their versions.
# If the plugin name contains a dot then you must add quotes to the key name!
[plugins]
"rocks.nvim" = "2.42.2"
"tokyonight.nvim" = "4.10.0"
...
"mason.nvim" = "1.10.0"
"mason-lspconfig.nvim" = "1.31.0"
```

*colorscheme.lua*
```
vim.cmd.colorscheme('tokyonight')
```

*dvorak.lua*
```
local map = vim.api.nvim_set_keymap
local options = { noremap = true, silent = true }

-- character movement (8)

...  
map('', 'S', 'g_l', options)         

-- word movement (8)

...
map('', 'L', 'E', options)
...
```

**Comment by Folaht** [comment, marked as answer]
Although there's a lot of complaints about 
https://mrcjkb.dev/posts/2023-08-22-setup.html
from the main developer of rocks, 
I haven't seen any alternative over this setup yet.

```
~/.config/nvim
├── init.lua
├── plugin
│   ├── colorscheme.lua
│   ├── keymap
│   │   └── dvorak.lua
│   └── plugins.lua
└── rocks.toml
```

`plugins.lua`
```
-- LSP settings
require('mason').setup {}
require('mason-lspconfig').setup()
```

**Reply by mrcjkb** [reply]
Yep, that article was directed at plugin authors, not users :smile: 
Many plugins still force their users to call `setup`.
